backend/apps/readings/analysis.py

```diff
@@ -473,3 +517,49 @@ def percent_using_relevant_words_by_context_and_question(self):
             question_context_percent_list.append((item[0], item[1], item[2] / total_student_count))
 
         return question_context_percent_list
+
+    def clean_resp_strings(self):
+        """
+        Removes punctuation from responses in dataset and makes all characters lowercase.
+        :return: a copy of dataset with no punctuation and all characters lowercase in the
+        value keyed by 'response'
+        """
+        clean_dataset = []
+        for resp in self.responses:
+            clean_dataset.append(resp)
+
+        for clean_resp in clean_dataset:
+            clean_resp.response = clean_resp.response.replace('.', '').lower()
+            clean_resp.response = clean_resp.response.replace(',', '')
+
+        return clean_dataset
+
+    def extract_responses_by_context(self):
+        """
+        Extracts text responses to "In three words or fewer, what is this text about?" from given
+        csv
+        file, depending on context given (story vs. ad). Returns lists of words used in question
+        responses. Also removes duplicates.
+        :return: response_ad and response_story, lists of strings of all words used in user
+        responses
+        (duplicates included)
+        """
+        # cycling through dataset to find prompt responses, sorting based on context
+        responses_ad = []
+        responses_story = []
+        question = "In three words or fewer, what is this text about?"
+
+        # removing punctuation and making lowercase
+        clean_dataset = self.clean_resp_strings()  # removing punctuation and making lowercase
+        for resp in clean_dataset:
```

Comment:
rather than hardcoding the contexts and Qs, can you iterate over all contexts and questions? this will make this more usable as we move into the final analysis of responses to arbitrary questions posed by an instructor.